src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
Does this relate to the fix that we are looking at or is it a general cleanup? 
The comment is correct, it never said that ContainsPrefixAsync does not return true if an exact match exists. 
Complex objects can be directly bound by the value providers, unless there is a strong reason for removing this i would like to keep this. 
> works better in some scenarios.

Can you point me to those scenarios?